mmv1/products/vertexai/IndexEndpointDeployedIndex.yaml

```diff
@@ -0,0 +1,264 @@
+        output: true
+      - !ruby/object:Api::Type::Array
+        name: 'pscAutomatedEndpoints'
+        description: |
+          PscAutomatedEndpoints is populated if private service connect is enabled if PscAutomatedConfig is set.
+        output: true
+        item_type: !ruby/object:Api::Type::NestedObject
+          properties:
+            - !ruby/object:Api::Type::String
+              name: projectId
+              output: true
+              description: |
+                Corresponding projectId in pscAutomationConfigs
+            - !ruby/object:Api::Type::String
+              name: network
+              output: true
+              description: |
+                Corresponding network in pscAutomationConfigs.
+            - !ruby/object:Api::Type::String
+              name: 'matchAddress'
+              output: true
+              description: |
+                ip Address created by the automated forwarding rule.
+  - !ruby/object:Api::Type::String
+    name: 'indexSyncTime'
+    output: true
+    description: If this timestamp's value is at least the Index.update_time of the original Index, it means that this DeployedIndex and the original Index are in sync. If this timestamp is older, then to see which updates this DeployedIndex already contains (and which it does not), one must list the operations that are running on the original Index.
+  - !ruby/object:Api::Type::NestedObject
+    name: automaticResources
+    description: A description of resources that the DeployedIndex uses, which to large degree are decided by Vertex AI, and optionally allows only a modest additional configuration.
+    default_from_api: true
+    properties:
+      - !ruby/object:Api::Type::Integer
+        name: 'minReplicaCount'
+        description: The minimum number of replicas this DeployedModel will be always deployed on.
+        default_from_api: true
+      - !ruby/object:Api::Type::Integer
+        name: 'maxReplicaCount'
+        description: The maximum number of replicas this DeployedModel may be deployed on when the traffic against it increases. If maxReplicaCount is not set, the default value is minReplicaCount
+        default_from_api: true
+  - !ruby/object:Api::Type::NestedObject
+    name: dedicatedResources
+    description: A description of resources that are dedicated to the DeployedIndex, and that need a higher degree of manual configuration.
+    properties:
+      - !ruby/object:Api::Type::NestedObject
+        name: 'machineSpec'
+        description: The minimum number of replicas this DeployedModel will be always deployed on.
+        immutable: true
+        required: true
+        properties:
+          - !ruby/object:Api::Type::String
+            name: 'machineType'
+            description: The type of the machine.
+            immutable: true
+      - !ruby/object:Api::Type::Integer
+        name: 'minReplicaCount'
```

Comment:
The [docs describe this field as immutable](https://cloud.google.com/vertex-ai/docs/reference/rest/v1/DedicatedResources) FYI!

Reply:
They mention minReplicaCounts and maxReplicaCounts as being the fields that are mutatable but don't specifiy which ones (automatic/dedicated), so in this case the truly only updatable fields are from `automaticResources`.

https://cloud.google.com/vertex-ai/docs/vector-search/deploy-index-vpc#mutate-deployed-index

Reply:
This is actually something that was brought up with riley and is also mentioned in a previous comment in this PR, because it's a fine grained resource the docs won't really match with what actual process is. I took another look at the docs and they have both automatic/dedicated fields being immutable which would go against the ability to mutate the resource. This is also mentioned in this comment here: https://github.com/GoogleCloudPlatform/magic-modules/pull/11039#issuecomment-2209316648